src/Input/TypeHintedParamAwareInput.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-cli for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-cli/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-cli/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Cli\Input;
+
+use InvalidArgumentException;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\StreamableInputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+use function is_array;
+use function sprintf;
+
+/**
+ * Decorate an input instance to add a `getParam()` method.
+ *
+ * Compatible with symfony/console 5.0+.
+ *
+ * @internal
+ */
+final class TypeHintedParamAwareInput implements ParamAwareInputInterface
+{
+    /** @var QuestionHelper */
+    private $helper;
+
+    /** @var InputInterface */
+    private $input;
+
+    /** @var OutputInterface */
+    private $output;
+
+    /** @var array array<string, InputParamInterface> */
+    private $params;
+
+    public function __construct(InputInterface $input, OutputInterface $output, QuestionHelper $helper, array $params)
+    {
+        $this->input  = $input;
+        $this->output = $output;
+        $this->helper = $helper;
+        $this->params = $params;
+    }
+
+    /**
+     * @return null|mixed
+     * @throws InvalidArgumentException When the parameter does not exist.
+     * @throws InvalidArgumentException When the parameter is of an invalid type.
+     * @throws InvalidArgumentException When the parameter is required, input is
+     *     non-interactive, and no value is provided.
+     */
+    public function getParam(string $name)
+    {
+        if (! is_array($this->params) || ! isset($this->params[$name])) {
```

Comment:
Quite a big part of this method body is duplicated. Maybe we should extract it to internal class and reuse in both places?